pyriemann/utils/covariance.py

```diff
@@ -151,21 +180,47 @@ def cospectrum(X, window=128, overlap=0.75, fmin=None, fmax=None, fs=None):
         fdata[window_ix, :, :] = numpy.fft.fft(
             cdata, n=window, axis=1)[:, 0:number_freqs]
 
-    # Adjust Frequency range to specified range (in case it is a parameter)
-    if fmin is not None:
+    # adjust frequency range to specified range (in case it is a parameter)
+    if fs is not None:
+        if fmin is None:
+            fmin = 0
+        if fmax is None:
+            fmax = fs / 2
+        if fmax <= fmin:
+            raise ValueError('Parameter fmax must be superior to fmin')
+        if 2.0 * fmax > fs: # check Nyquist-Shannon
+            raise ValueError('Parameter fmax must be inferior to fs/2')
         f = numpy.arange(0, 1, 1.0 / number_freqs) * (fs / 2.0)
         Fix = (f >= fmin) & (f <= fmax)
         fdata = fdata[:, :, Fix]
+        freqs = f[Fix]
+    else:
+        if fmin is not None:
+            raise Warning('Parameter fmin not used because fs is None')
+        if fmax is not None:
+            raise Warning('Parameter fmax not used because fs is None')
```

Comment:
the standard way is to do:

```python
import warnings
warnings.warn("Parameter fmax not used because fs is None")
```

Reply:
unless you want to reallly raise an Exception in which case I would suggest a ValueError